variation-commons-mongodb/src/main/java/uk/ac/ebi/eva/commons/mongodb/repositories/VariantRepositoryImpl.java

```diff
@@ -108,6 +110,28 @@ public Long countByRegionsAndComplexFilters(List<Region> regions, List<VariantRe
                 .distinct(VariantMongo.CHROMOSOME_FIELD));
     }
 
+    @Override
+    public List<VariantMongo> findByRegionAndOtherBeaconFilters(Region startRange, Region endRange,
+                                                                List<VariantRepositoryFilter> filters) {
+        Query query = new Query();
+        query.addCriteria(Criteria.where(VariantMongo.CHROMOSOME_FIELD).is(startRange.getChromosome()));
+
+        Criteria startCriteria = getRegionRangeCriteria(startRange, VariantMongo.START_FIELD);
+        Criteria endCriteria = getRegionRangeCriteria(endRange, VariantMongo.END_FIELD);
+        if (startCriteria != null) {
+            query.addCriteria(startCriteria);
+        }
+
+        if (endCriteria != null) {
+            query.addCriteria(endCriteria);
+        }
+
+        pageable_flag=false;
+        List<VariantMongo> variantMongoList = findByComplexFiltersHelper(query, filters, null, null);
+        pageable_flag=true;
```

Comment:
I would advise against adding complexity to the state of this class such as field=true; method(); field=false; I think a better approach is to take a Pageable parameter in this function too (as in any other find... function in this class.

In eva-ws we will have to provide the Pageable and decide how to make sure we are not missing any record

Reply:
As discussed on slack, I have created new pageable object for the querying.